### PR TITLE
Make RotState reflect hubris state better

### PR DESCRIPTION
**Pull request by andrewjstone**
This matches the changes in [hubris#1009](https://github.com/oxidecomputer/hubris/pull/1009) but should merge first, once the hubris PR is ready for merge.

### CODE REVIEW

File: gateway-messages/src/sp_to_mgs.rs [hunk 1]
```diff
@@ -149,13 +149,42 @@ pub struct SpState {
 #[derive(
     Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
 )]
-pub struct RotState {
+pub enum RotSlot {
+    A = 0,
```
Comment (jgallagher) [inline gateway-messages/src/sp_to_mgs.rs:153]: What do we gain from the explicit `A = 0`, `B = 1`?

File: gateway-messages/src/sp_to_mgs.rs [hunk 2]
```diff
@@ -149,13 +149,42 @@ pub struct SpState {
...
+    pub slot_a: Option<RotImageDetails>,
+    pub slot_b: Option<RotImageDetails>,
+}
+
+// TODO(AJS): Fill in with runtime state - i.e. updates that have completed before an RoT reset
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub struct RotUpdateDetails {
+    pub boot_state: RotBootState,
```
Comment (jgallagher) [inline gateway-messages/src/sp_to_mgs.rs:180]: Should `boot_state` be a member of `RotState` instead? This question is probably coming mostly from the fact that both `RotState` and `RotUpdateDetails` only have one member each; maybe you have ideas in mind for how each will be fleshed out and why this is placed where it is?

File: gateway-messages/src/sp_to_mgs.rs (hunk 1 above)
Comment (andrewjstone) [reply gateway-messages/src/sp_to_mgs.rs:153]: Nothing! I will remove it.

File: gateway-messages/src/sp_to_mgs.rs (hunk 2 above)
Comment (andrewjstone) [reply gateway-messages/src/sp_to_mgs.rs:180]: It's a good question. There's internally an RotUpdateDetails in hubris that is eventually going to contain the RotBootState and some runtime information as described in https://github.com/oxidecomputer/hubris/issues/1021. I was just leaving a placeholder for that. If you'd prefer I not do that I can change it for now. Either way is fine with me.

Comment (jgallagher) [reply gateway-messages/src/sp_to_mgs.rs:180]: I don't have a strong preference either way if we're going to be changing it more soon anyway. I'm fine with leaving it as-is, just wanted to ask the question.

**Review by jgallagher — APPROVED** [review]

**Pull request merged**